packages/components/src/lib/d3/patchedScales.ts

```diff
@@ -19,9 +19,9 @@ function squiggleDefaultFormat() {
     currency: ["$", ""],
     minus: "-",
   });
-  const siFormat = locale.format(".9~s");
-  const expFormat = locale.format(".9~e");
-  const fixedFormat = locale.format(".9~f");
+  const siFormat = locale.format(".3~s");
+  const expFormat = locale.format(".3~e");
+  const fixedFormat = locale.format(".3~f");
```

Comment:
This causes a blind spot in 0.001..0.00001 interval: https://squiggle-website-git-plot-additions-quantified-uncertainty.vercel.app/playground?v=dev#code=eNqrVkpJTUsszSlxzk9JVbJSMtAzAAJDhZJ8BTDTUElHqTgjvzy4NDc3sahSyaqkqDS1FgC78hDF

This is a problem particularly for relative values where dists like this are going to be frequent.

It's especially confusing if you try to move your cursor from 0 to some value around 0.001:
- first you get tiny non-zero values (1e-6)
- then you get to this blind spot where verything is 0
- and then from 0.001 it's non-zero again

`.3~r` works well for values under `0.001`:

```
> d3.format('0.3~f')(0.03123123)
'0.031'
> d3.format('0.3~f')(0.0003123123)
'0'
> d3.format('0.3~r')(0.03123123)
'0.0312'
> d3.format('0.3~r')(0.0003123123)
'0.000312'
```

Then above 1000 it becomes bad again:
```
> d3.format('0.3~r')(12345.123145125)
'12300'
```

How about another `else if` branch below that uses `.3~r` for values below 0.001 and `.3~f` above that?

It's unfortunate that there's nothing in d3 for "show all digits before decimal point, and first N significant digits after that".

Reply:
Good idea about the ``else if`` branch, will try.

I was nervous about this, but was figuring I'd address later, but happy to do it now.

Reply:
I just reverted this for this story, will move to a story.